pkg/runtime/runtime.go

```diff
@@ -157,6 +159,14 @@ func (r *Runtime) getEnv(inherit bool) (map[string]string, map[string]string, er
 		return empty, empty, errs.Wrap(err, "Failed to get environment variables")
 	}
 
+	if ignores := os.Getenv(constants.IgnoreEnvEnvVarName); ignores != "" {
+		for _, name := range strings.Split(ignores, ",") {
+			if _, exists := vars[name]; exists {
+				delete(vars, name)
+			}
+		}
+	}
```

Comment:
This is deleting environment variables regardless of whether we created them. If inherit is true then you might be deleting the very PYTHONPATH the user was trying to set, along with ours.

I think we need to address this inside the envdef package. And it should prevent defining the env var, rather than delete it after the fact (unless we can do it without inheritance playing a role).

Reply:
No, there's an integration test that verifies we're leaving the user's set PYTHONPATH alone. This deletion only affects the runtime environment, which has not yet been merged with the current one. Since this runtime environment does not contain PYTHONPATH, when merging with the current one, the user's set PYTHONPATH will remain.

I tried to make this happen in envdef, but there are at least three different places where environment mutations happen, and all would need the check:

- https://github.com/ActiveState/cli/blob/d893ec6d386118897fdf4ce87dde6931687c87d5/pkg/runtime/internal/envdef/environment.go#L178-L222
- https://github.com/ActiveState/cli/blob/d893ec6d386118897fdf4ce87dde6931687c87d5/pkg/runtime/internal/envdef/environment.go#L370-L379
- https://github.com/ActiveState/cli/blob/d893ec6d386118897fdf4ce87dde6931687c87d5/pkg/runtime/internal/envdef/environment.go#L337-L362

I thought it would be better to capture it at the top-level than all those low-level places where things can slip through the cracks if we're not careful.

Reply:
I agree having a central place for it would be ideal, but this still feels too error prone to me. The fact that it passes in your tests worries me more than that it settles my concern, because reading through this code suggests that it should not work.

If we are to place the logic here then there needs to be an understanding of why that test is passing, and the code should clearly communicate either via logic or comments that this is working with an awareness of inheritance.